sentry-android-core/src/main/java/io/sentry/android/core/performance/AppStartMetrics.java

```diff
@@ -284,43 +299,49 @@ private void checkCreateTimeOnMain(final @NotNull Application application) {
                   appStartProfiler = null;
                 }
               }
-              application.unregisterActivityLifecycleCallbacks(instance);
             });
   }
 
   @Override
   public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
-    // An activity already called onCreate()
-    if (!appLaunchedInForeground || onCreateTime != null) {
+    final long nowUptimeMs = SystemClock.uptimeMillis();
+    activeActivitiesCounter.incrementAndGet();
+
+    // the first activity determines the app start type
+    if (activeActivitiesCounter.get() == 1 && !firstDrawDone.get()) {
+      // If the app (process) was launched more than 1 minute ago, it's likely wrong
+      final long durationSinceAppStartMillis = nowUptimeMs - appStartSpan.getStartUptimeMs();
+      if (!appLaunchedInForeground || durationSinceAppStartMillis > TimeUnit.MINUTES.toMillis(1)) {
+        appStartType = AppStartType.WARM;
+      } else {
+        appStartType = savedInstanceState == null ? AppStartType.COLD : AppStartType.WARM;
+      }
+    }
+    appLaunchedInForeground = true;
```

Comment:
```suggestion
```
useless if called earlier

Reply:
IMHO still fine, once an Activity is created - no matter if we determine a cold or warm start, at the end of this method an Activity was created and thus the app is in foreground